From 6b491ad27ac3e26cb8caa15d25ffa62b1a6e34d3 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 17 Aug 2020 09:47:36 +0700
Subject: [PATCH 503/504] ARM: dts: iwg21m: Use 4-bit data bus mode for eMMC

On iWave RZ/G1H platform (iwg21m), MMC_DATA7 pin is shared with VI1_CLK.
This patch switches from using 8-bit to 4-bit data bus mode for eMMC to
avoid possible pin confliction.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index ba91fd2..778c24a 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -516,6 +516,11 @@
 		function = "i2c3";
 	};
 
+	mmc1_4_pins: mmc1_4 {
+		groups = "mmc1_data4", "mmc1_ctrl";
+		function = "mmc1";
+	};
+
 	mmc1_8_pins: mmc1_8 {
 		groups = "mmc1_data8", "mmc1_ctrl";
 		function = "mmc1";
@@ -620,11 +625,11 @@
 };
 
 &mmcif1 {
-	pinctrl-0 = <&mmc1_8_pins>;
+	pinctrl-0 = <&mmc1_4_pins>;
 	pinctrl-names = "default";
 
 	vmmc-supply = <&reg_3p3v>;
-	bus-width = <8>;
+	bus-width = <4>;
 	non-removable;
 	status = "okay";
 };
-- 
2.7.4

